public/framework/core_refs.ts

```diff
@@ -41,3 +41,19 @@ class CoreRefs {
 }
 
 export const coreRefs = CoreRefs.Instance;
+
+/**
+ * Safely prepend the `basePath` from `coreRefs` to the given link.
+ * If `coreRefs.http.basePath` exists (always true in normal operation), prepend it to the link.
+ * If it doesn't exist (usually during unit testing), return the link as-is.
+ *
+ * @param link The link to prepend with `coreRefs.http.basePath`.
+ * @returns The link with the prepended `basePath` if it exists, otherwise the unmodified link.
+ */
+export const basePathLink = (link: string): string => {
```

Comment:
I think this is a worthwhile helper method but I'm not sure if it should go in core_refs, is there a better location?

Reply:
Maybe add it in `https://github.com/opensearch-project/dashboards-observability/blob/main/common/utils/shared.ts`As this one is not manipulating the coreRef itself rather giving a utility function over it.

Reply:
Moved it to shared [after fighting TS for 30 minutes](https://github.com/opensearch-project/dashboards-observability/pull/1560/files#diff-289813aec7d9814eed64d99c67ee43cea23e572e4bcbbce01047d5c6c8c5839cR6-R18).